tests/test_utils.py

```diff
@@ -34,6 +35,16 @@ def test_update_config(caplog) -> None:
     assert updated_config["c"] == "something new"
 
 
+def test_update_plotting_config(process_scan_config: dict) -> None:
```

Comment:
I was thinking this might be paramterised along the lines of...


```suggestion
@pytest.mark.parametrize("image_name,core_set,title,zrange",
[("extracted_channel",False,"Raw Height",KeyError()),
 ("z_threshed",True,"Height Threshold",*id001) 
def test_update_plotting_config(process_scan_config: dict, image_name: str, core_set: bool, title:str) -> None:

    process_scan_config["plotting"] = update_plotting_config(process_scan_config["plotting"])
    assert process_scan_config["plotting"]["plot_dict"][image_name]["core_set"] == core_set
    assert process_scan_config["plotting"]["plot_dict"][image_name]["title"] == title

```

Could have more values tested but `z_threshed`  is one of only two variables that should have a value for `zrange` the others all have that `.pop()`'ed off. That said the value `*id001` looks kind of strange :thinking: 

Hope that makes sense, let me know if not.

Reply:
I believe I have successfully parameterized them now. The test runs successfully on my machine, and it's late so I'm going to cross my fingers and get some sleep.

Let me know if you're unhappy with the way I handled the `KeyError` exception, I didn't know how else I could handle it (using `KeyError()` as an argument didn't seem to work).

Thanks for your speedy feedback, it's really appreciated.